src/logic/tokens/store/actions/fetchTokens.js

```diff
@@ -50,6 +50,27 @@ export const containsMethodByHash = async (contractAddress: string, methodHash:
   return byteCode.indexOf(methodHash.replace('0x', '')) !== -1
 }
 
+let tokensInstanceCache = {}
+
+export const getTokenInstance = async (tokenAddress: string) => {
+  if (!tokenAddress) {
+    return null
+  }
+  let tokenInstance = tokensInstanceCache[tokenAddress]
+  // If the token is inside the cache we return the cached token
+  if (tokenInstance) {
+    return tokenInstance
+  }
+  // Otherwise we fetch it, save it to the cache and return it
+  const tokenContract = await getHumanFriendlyToken()
+  tokenInstance = await tokenContract.at(tokenAddress)
+  const [tokenSymbol, tokenDecimals] = await Promise.all([tokenInstance.symbol(), tokenInstance.decimals()])
+  tokenInstance.symbol = () => tokenSymbol
```

Comment:
Here you can use batching https://web3js.readthedocs.io/en/v1.2.0/web3-eth.html#id9 to do one query instead of 2 (`tokenSymbol` and `tokenDecimals` in the same query)

Reply:
Also, when you get call the `balance` or the `balanceUsd` endpoint in the transaction service you get the token, so you can have already some tokens cached from there :)